CMakeLists.txt

```diff
@@ -118,3 +118,14 @@ endif()
 if (CUB_ENABLE_EXAMPLES)
   add_subdirectory(examples)
 endif()
+
+# Documentation targets for CUB
+if (CUB_TOPLEVEL_PROJECT)
```

Comment:
We should put this behind a `CUB_ENABLE_DOXYGEN` option and rename the target to `cub.docs` so we can invoke this from Thrust builds.

The output path should be customizable and default to a location in the build dir instead of the source.

Reply:
addressed these review comments.